app/src/main/java/com/duckduckgo/app/browser/orientation/JsOrientationHandler.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.orientation
+
+import android.content.pm.ActivityInfo
+import androidx.fragment.app.FragmentActivity
+import com.duckduckgo.app.global.view.requestJsOrientationChange
+import com.duckduckgo.js.messaging.api.JsCallbackData
+import org.json.JSONObject
+
+class JsOrientationHandler {
+
+    /**
+     * Updates the activity's orientation based on provided JS data
+     *
+     * @return response data
+     */
+    fun updateOrientation(data: JsCallbackData, activity: FragmentActivity?): JsCallbackData {
+        val response = if (activity == null) {
+            NO_ACTIVITY_ERROR
+        } else {
+            val requestedOrientation = data.params.getString("orientation")
```

Comment:
This can crash the app if `orientation` doesn't exist.

Reply:
Yes, my JS-side unit tests uncovered this too! 😄  Will fix in the next round 👍